### PR TITLE
Add tests and CI

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Add a test module, with unit and integration tests (the unit test using the framework). Also adds shippable CI config.

##### ISSUE TYPE
- Bugfix Pull Request
- Feature Pull Request

##### COMPONENT NAME
all

### CODE REVIEW

**Comment by felixfontein** [comment]
Tests pass (locally with devel's ansible-test, haven't tried 2.9's ansible-test yet, but I guess that should also work).

File: tests/utils/shippable/shippable.sh [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+                    # remove numbered component from stub files when converting to tags
+                    flags="${flags//stub-[0-9]*/stub}"
+                    flags="${flags//=/,}"
+                    flags="${flags//[^a-zA-Z0-9_,]/_}"
+
+                    bash <(curl -s https://codecov.io/bash) \
+                        -f "${file}" \
+                        -F "${flags}" \
+                        -n "${test}" \
+                        -t 20636cf5-4d6a-4b9a-8d2d-6f22ebbaa752 \
```
Comment (gundalow) [inline tests/utils/shippable/shippable.sh:162]:
```suggestion
                        -t 2e861daf-bef9-4523-b544-4f25c46d491c \
```

**Comment by gundalow** [comment]
Excellent work :)

**Comment by felixfontein** [comment]
@gundalow @resmo thanks a lot!

**Pull request merged**